src/theia/sfm/feature_extractor_and_matcher.cc

```diff
@@ -63,8 +63,24 @@ void ExtractFeatures(
     const FeatureExtractorAndMatcher::Options& options,
     const std::string& image_filepath,
     std::vector<Keypoint>* keypoints,
-    std::vector<Eigen::VectorXf>* descriptors) {
+    std::vector<Eigen::VectorXf>* descriptors,
+    const std::string& imagemask_filepath = "") {
   std::unique_ptr<FloatImage> image(new FloatImage(image_filepath));
+  std::unique_ptr<FloatImage> image_mask(new FloatImage(imagemask_filepath));
+
+  if (imagemask_filepath != "") {
+  // Check the size of the image and its associated mask.
+    if (image_mask.get()->Width() != image.get()->Width() ||
```

Comment:
You can do CHECK(image_mask->Width() == image->Width() && image_mask->Height() == image->Height()) << "Error message";